### PR TITLE
Enable auth on etcd and pass two more certs

### PR DESCRIPTION
**Pull request by lubronzhan**
https://www.pivotaltracker.com/story/show/163786805

### CODE REVIEW

File: jobs/etcd/spec [hunk 1]
```diff
@@ -56,9 +58,29 @@ properties:
     description: Certificate for etcdctl client authentication
   tls.etcdctl.private_key:
     description: Private key for etcdctl client authentication
+  tls.etcdctl-flanneld.certificate:
```
Comment (tvs) [inline jobs/etcd/spec]: Do we even need these? Flannel currently consumes certs via the `etcd` link, but given that `etcd` doesn't appear to need to know of these (based on the property not being used), there's no reason we couldn't just pass these creds directly to Flannel.

File: jobs/etcd/spec [hunk 2]
```diff
@@ -56,9 +58,29 @@ properties:
...
+    description: Certificate for etcdctl used by flanneld job client authentication
+  tls.etcdctl-flanneld.private_key:
+    description: Private key for etcdctl used by flanneld job client authentication
   tls.peer.ca:
     description: CA for peer authentication
   tls.peer.certificate:
     description: Certificate for peer authentication
   tls.peer.private_key:
     description: Private key for peer authentication
+  users:
```
Comment (tvs) [inline jobs/etcd/spec]: This list should probably let you choose the API version to create the user under - `flanneld` for example needs to be created in `v2`, but root probably needs to exist on both `v2` and `v3`.

e.g.

```
- name: root
  password: ((etcd_user_root_password))
  versions: [ "v2", "v3" ]
```

File: jobs/etcd/templates/bin/post-start.erb [hunk 3]
```diff
@@ -3,17 +3,108 @@
...
+    member_id="$($etcdctl member list | grep "<%= spec.id %>" | cut -d',' -f1)"
+
+    echo "Updating Member (${member_id}) peer url"
+
+    $etcdctl member update "${member_id}" --peer-urls "${etcd_peer_address}"
+  fi
+}
+
+setup_users_auth() {
+  export ETCDCTL_API=2
```
Comment (tvs) [inline jobs/etcd/templates/bin/post-start.erb]: As with above, this should probably be configurable per-user.

File: jobs/etcd/spec (hunk 1 above)
Comment (lubronzhan) [reply jobs/etcd/spec]: Cool let me try

File: jobs/etcd/spec (hunk 2 above)
Comment (lubronzhan) [reply jobs/etcd/spec]: Well the problem is the etcdctl command for v2 and v3 have big difference. But yeah let me try then.

File: jobs/etcd/templates/bin/authentication-setup.erb [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+}
+
+setup_users_auth() {
+  <% if_p('users') do |users| %>
+  versions=""
+  <% users.each do |user| %>
+    <% if user["versions"] != nil %>
+    <% user["versions"].each do |version| %>
+      <% if version == "v3" %>
+  export ETCDCTL_API=3
```
Comment (tvs) [inline jobs/etcd/templates/bin/authentication-setup.erb]: Any chance we can do this without mucking with the exported ENV? It's akin to using a global variable and makes it a little difficult to track what's going on here.

**Pull request merged**